Add `#include <limits>` in all files where it is used

hyarion opened the pull request:
There where some environments that couldn't build due to missing includes, this adds it where used (IWYU style)

chennes commented:
I am a strong proponent of "include what you use", having been bit by the problem simply when upgrading to a new version of the same compiler, etc. +1 from me.

furgo16 commented:
Thanks, building now at https://github.com/furgo16/FreeCAD-snap/actions/runs/14187503692

furgo16 commented:
The snap build succeeded (though it doesn't include the latest commit https://github.com/FreeCAD/FreeCAD/pull/20559/commits/bad0daf9dc346e9d59c0cf715afc1b67cd9d86d4, as the build started before it was added).

benj5378 commented:
The issue here is, that previously includes that were indirectly included, was removed. Are we just adding and removing headers reoccuringly?

hyarion commented:
> The issue here is, that previously includes that were indirectly included, was removed. Are we just adding and removing headers reoccuringly?

The previous version (pre std::numeric_limits) relied on indirect includes. When I refactored it, I removed the old includes I could find and added enough `#include <limits>` to make it pass CI. I did this to keep the change set down and since old version relied on indirect imports I thought it was good enough to do the same here. However, it seems like drmacro's and the snap builds are built differently so it caused them problems.

I did include some minor boy scout fixes that I encountered. The biggest change is probably the last commit (and is why I put it in a separate commit).

benj5378 commented:
I don't think this is due to incorrect includes. I think it is due to some GCC bug, that are only fixed in some environments:
```cpp
/tmp/freecad-aur/src/FreeCAD/src/Tools/plugins/widget/customwidgets.cpp:1524:24: error: ‘auto’ not allowed in alias declaration
 1524 |         using limits = std::numeric_limits;
      |   
```
https://gcc.gnu.org/bugzilla/show_bug.cgi?id=104091

I think we might fix the issue by changing
```cpp
using limits = std::numeric_limits;
```
to
```cpp
template<typename T...>
using limits = std::numeric_limits<T...>;
```
